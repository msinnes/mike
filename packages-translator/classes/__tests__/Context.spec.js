const { loadClass } = require('@mike/class');

const Context = require('../Context');

describe('Context', () => {
  it('should be an abstract class', () => {
    expect(() => {
      Context('string');
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new Context('string');
    }).toThrowErrorMatchingSnapshot();
  });

  describe('expose', () => {
    it('should have an expose method', () => {
      expect(Context.expose).toBeInstanceOf(Function);
    });

    it('should assign a property onto an object', () => {
      const valueRef = {};
      const TestableContext = loadClass(function (value) {
        TestableContext.expose(this, 'fieldName', () => value);
      }).extend(Context);
      const instance = new TestableContext(valueRef);
      expect(instance.fieldName).toEqual(valueRef);
    });

    it('should not be writable', () => {
      const valueRef = {};
      const TestableContext = loadClass(function (value) {
        TestableContext.expose(this, 'fieldName', () => value);
      }).extend(Context);
      const instance = new TestableContext(valueRef);
      instance.fieldName = 'something else';
      expect(instance.fieldName).toEqual(valueRef);
    });
  });
});