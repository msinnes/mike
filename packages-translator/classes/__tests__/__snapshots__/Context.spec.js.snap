// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Context should be an abstract class 1`] = `"ClassError: Cannot call a class as a function"`;

exports[`Context should be an abstract class 2`] = `"ClassError: Abstract classes cannot be instantiated"`;
